pkg/asset/machines/azure/azuremachines.go

```diff
@@ -144,6 +145,34 @@ func GenerateMachines(clusterID, resourceGroup, subscriptionID string, in *Machi
 		}
 	}
 
+	defaultDiag := &capz.Diagnostics{
+		Boot: &capz.BootDiagnostics{
+			StorageAccountType: capz.ManagedDiagnosticsStorage,
+		},
+	}
```

Comment:
Default machine pool values are setup here:

https://github.com/openshift/installer/blob/d6a6613503a1c1907e34898df43c5d72ba600c68/pkg/asset/machines/worker.go#L117-L125

If you think it would simplify the implementation you can move the default settings here. It would reduce code across the three files. You would need  to extend the function to accept the role name and conditionally set control plane vs compute.